### ISSUE TITLE
sqlite3: Add configure option to set or auto-detect rpath to sqlite3 libs

### ISSUE THREAD

**Issue by 99ecf41b-94db-4cd4-b6ba-d77b8869c099** [python/cpython#89537]
BPO | [45374](https://bugs.python.org/issue45374)
--- | :---
Nosy | @erlend-aasland

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2021-10-05.09:47:10.027>
labels = ['type-feature', 'expert-installation', 'build', '3.11']
title = 'sqlite3: Add configure option to set or auto-detect rpath to sqlite3 libs'
updated_at = <Date 2021-10-05.21:17:34.559>
user = 'https://bugs.python.org/chaz6'
```

bugs.python.org fields:
```python
activity = <Date 2021-10-05.21:17:34.559>
actor = 'erlendaasland'
assignee = 'docs@python'
closed = False
closed_date = None
closer = None
components = ['Build', 'Installation']
creation = <Date 2021-10-05.09:47:10.027>
creator = 'chaz6'
dependencies = []
files = []
hgrepos = []
issue_num = 45374
keywords = []
message_count = 1.0
messages = ['403214']
nosy_count = 3.0
nosy_names = ['chaz6', 'docs@python', 'erlendaasland']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'enhancement'
url = 'https://bugs.python.org/issue45374'
versions = ['Python 3.11']
```

</p></details>

**Comment by erlend-aasland** [comment]
Quoting the other Ned in https://github.com/python/cpython/pull/95547#issuecomment-1202025158:

> Without giving it very much thought, it seems to me that adding yet another explicit configure option for one specific library is a step in the wrong direction for 3.12 and beyond. With all the good work in 3.11 and 3.12 to provide consistent ways to override configuration options for all third-party libraries, I think it would be worthwhile to give some thought as to how this could be done for RPATH options as well. I can't think of any reasons why we should provide special options for just openssl and sqlite3 libraries; sooner or later someone will have a legitimate request for RPATH options for others.

As I already replied on the PR, this is a valid concern. We should try to find a way to do this for all linked libraries (bzip2, uuid, Tcl/Tk, etc.). I think the PR takes us one step in the right direction, with the addition of the rpath helper macro[^1].

cc. @tiran 

[^1]: The macro will probably need further tweaking, though.

**Comment by tiran** [comment]
I agree with Erlend, if we decide to add an option, then it should be a generic option that applies to all libraries.

**Comment by erlend-aasland** [comment]
Possible solutions, from the top of my head, before breakfast coffee has been consumed:

1. One option for each module, but only documented once:
   ```
   $ ./configure --help | grep rpath
   --with-rpah-<mod>=[DIR|auto|no]
   ```
2. Single option, values provided as list:
   ```
   $ ./configure --with-rpath=sqlite3:no,openssl:auto,spam:/usr/local/lib/eggs
   ```
3. Single option, allow it to be specified multiple times:
   ```
   $ ./configure --with-rpath=sqlite3:no --with-rpath=openssl:auto --with-rpath=spam:/usr/local/lib/eggs
   ```

**Comment by tiran** [comment]
It's a convenience option. I would keep it simple and just add an option like `--enable-auto-rpath` that adds an rpath for every `-L` options for each and every module. If a user requires more control over rpath, they can use env var overrides instead.

**Comment by erlend-aasland** [comment]
Yes, that's perhaps the best option. That also means minimal changes to configure.
